### PR TITLE
FISH-666 FISH-1372 MicroProfile 4.1

### PR DESCRIPTION
**Pull request by Pandrex247**
## Description
New Feature/Improvement.

Implements MicroProfile 4.1, which consists of just MicroProfile Health 3.1.

MicroProfile Health 3.1 introduces a `@Startup` annotation.

Also increments JWT-Auth for 1.2.1, which appears to just be a patch update with no changes - something presumably wrong with the release of 1.2.

### Blockers

https://github.com/payara/MicroProfile-TCK-Runners/pull/147

### Related

https://github.com/payara/MicroProfile-TCK-Runners/pull/152
https://github.com/payara/Payara/pull/5372

### New tests
None.

### Testing Performed
Ran all of the MicroProfile TCKs locally.

### Testing Environment
Windows 10, JDK 8.
WSL OpenSUSE Leap 15.2, JDK 8.

## Documentation
https://github.com/payara/Payara-Community-Documentation/pull/170

## Notes for Reviewers
None.

### CODE REVIEW

File: appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/HealthCheckService.java [hunk 1]
```diff
@@ -369,16 +380,20 @@ public void registerClassLoader(String appName, ClassLoader classloader) {
...
+            // Make sure we do a deep-copy first, otherwise the first map the foreach consumer gets used on will be a
+            // shallow copy: the two maps will essentially be the same map (changes to one affecting the other)
+            healthChecks = readiness.entrySet().stream().collect(Collectors.toMap(entry ->
+                    entry.getKey(), entry -> new HashSet(entry.getValue())));
             BiConsumer<? super String, ? super Set<HealthCheck>> mergeHealthCheckMap
                     = (key, value) -> healthChecks.merge(key, value, (oldValue, newValue) -> {
                         oldValue.addAll(newValue);
                         return oldValue;
                     });
-            //FIXME most likely this unintentionally changes the fields
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/HealthCheckService.java:379]: Presumably they were referring to the deep vs. shallow copy problem

**Comment by rdebusscher** [comment]
Is using the newly decided /health/started of RC3!  👍

**Review by lprimak — APPROVED** [review]

**Comment by lprimak** [comment]
jenkins test

**Comment by lprimak** [comment]
build fails... MP Health? 3.1 is not out yet I guess :)

**Comment by Pandrex247** [comment]
Correct.
Current target release date is the 7th of July: https://projects.eclipse.org/projects/technology.microprofile/reviews/4.1-release-review

**Comment by Pandrex247** [comment]
Rebased ontop of https://github.com/payara/Payara/pull/5318

**Comment by Pandrex247** [comment]
Jenkins test please

**Pull request merged**